### ISSUE TITLE
[Bug]: [metric]flowgraph buffer size value is a negative number

### ISSUE THREAD

**Issue by zhuwenxing** [milvus-io/milvus#32403]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:2.4-20240417-8f7ac8f7-amd64
- Deployment mode(standalone or cluster):
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

![image](https://github.com/milvus-io/milvus/assets/12268675/f8d6be62-5675-4a1c-a414-14d66dc05864)

**Comment by bigsheeper** [comment]
Related to the `writeBuffer`, and I haven't detected any suspicious aspects. Please help to look into this issue. @congqixia

**Comment by congqixia** [comment]
patch merged, could you please verify? 
/assign @zhuwenxing

**Comment by zhuwenxing** [comment]
verified and fixed in `master-20240426-bed6363f-amd64`
![image](https://github.com/milvus-io/milvus/assets/12268675/9a41c5db-8f19-464b-a6f0-692329486a73)
